src/main/java/transact/logic/commands/EditTransactionCommand.java

```diff
@@ -0,0 +1,244 @@
+package transact.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static transact.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static transact.logic.parser.CliSyntax.PREFIX_DATE;
+import static transact.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static transact.logic.parser.CliSyntax.PREFIX_STAFF;
+import static transact.logic.parser.CliSyntax.PREFIX_TYPE;
+import static transact.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import transact.commons.core.index.Index;
+import transact.commons.util.CollectionUtil;
+import transact.commons.util.ToStringBuilder;
+import transact.logic.Messages;
+import transact.logic.commands.exceptions.CommandException;
+import transact.model.Model;
+import transact.model.person.Person;
+import transact.model.transaction.Transaction;
+import transact.model.transaction.info.Amount;
+import transact.model.transaction.info.Date;
+import transact.model.transaction.info.Description;
+import transact.model.transaction.info.TransactionId;
+import transact.model.transaction.info.Type;
+import transact.ui.MainWindow.TabWindow;
+
+
+
+
+/**
+ * Edits the details of an existing person in the address book.
+ */
+public class EditTransactionCommand extends Command {
+
+    public static final String COMMAND_WORD = "edittransaction";
```

Comment:
```suggestion
    public static final String COMMAND_WORD = "edit";
```
I think we can shorten this so the user doesn't have to type so much, same for addtransaction

Reply:
Mmm maybe can `edit t` or `edit s` like `view`